source/MaterialXCore/Document.cpp

```diff
@@ -208,9 +208,49 @@ NodeDefPtr Document::addNodeDefFromGraph(const NodeGraphPtr nodeGraph, const str
         }
     }
 
-    for (auto output : graph->getOutputs())
+    // Expose any existing interface.
+    // Any connection attributes on the existing interface should be removed from the definition.
+    // as well as any source URI
+
+    // Attributes which should not be copied over
```

Comment:
Logic to copy over inputs properly, and strip out things like connection attributes for both inputs and outputs.